frontend/components/Layout/User.tsx

```diff
@@ -0,0 +1,128 @@
+import {
+   Avatar,
+   Flex,
+   Icon,
+   IconProps,
+   Link,
+   Menu,
+   MenuButton,
+   MenuDivider,
+   MenuGroup,
+   MenuItem,
+   MenuList,
+   Text,
+   VStack,
+} from '@chakra-ui/react';
+import { IFIXIT_ORIGIN } from '@config/env';
+import { useAuthenticatedUser } from '@lib/api';
+import * as React from 'react';
+
+export function UserMenu() {
+   const { user } = useAuthenticatedUser();
+
+   return (
+      <Flex>
+         {user ? (
+            <Menu>
+               <MenuButton
+                  aria-label="Open user menu"
+                  borderRadius="full"
+                  _focus={{
+                     boxShadow: 'outline',
+                     outline: 'none',
+                  }}
+               >
+                  <Avatar
+                     name={user.username}
+                     src={user.thumbnail || undefined}
+                     size="sm"
+                  />
+               </MenuButton>
+               <MenuList color="black">
+                  <VStack align="flex-start" px="0.8rem" py="0.4rem">
+                     <Text color="trueGray.900" fontWeight="semibold">
+                        {user.username}
+                     </Text>
+                     <Text color="trueGray.700">@{user.handle}</Text>
+                  </VStack>
+                  <MenuDivider />
+                  <MenuGroup>
+                     <MenuItem
+                        as="a"
+                        href={`${IFIXIT_ORIGIN}/User/Notifications/${user.id}/${user.username}`}
+                        fontSize="sm"
+                     >
+                        Notifications
+                     </MenuItem>
+                     <MenuItem
+                        as="a"
+                        fontSize="sm"
+                        href={`${IFIXIT_ORIGIN}/User/${user.id}/${user.username}`}
+                     >
+                        View Profile
+                     </MenuItem>
+                     <MenuItem
+                        as="a"
+                        fontSize="sm"
+                        href={`${IFIXIT_ORIGIN}/User/Orders`}
+                     >
+                        Orders
+                     </MenuItem>
+                  </MenuGroup>
+                  <MenuDivider />
+                  <MenuGroup>
+                     <MenuItem
+                        as="a"
+                        fontSize="sm"
+                        href={`${IFIXIT_ORIGIN}/Guide/logout`}
+                     >
+                        Log Out
+                     </MenuItem>
+                  </MenuGroup>
+               </MenuList>
+            </Menu>
+         ) : (
+            <Link borderRadius="full" href={`${IFIXIT_ORIGIN}/login`}>
+               <NoUserIcon
+                  transition="color 300ms"
```

Comment:
If we're trying to use 300ms transition duration across components, maybe we should add it to the chakra theme?